api.js

```diff
@@ -128,6 +128,28 @@ exports.init = function (sbot, config) {
     }
   }
 
+  function findOrCreateV1(root, cb) {
+    findOrCreate(root, v1Visit, v1Details, cb)
+  }
+
+  function findShard(v1Feed, root, details, cb) {
+    const shardDetails = {
+      feedpurpose: pickShard(root.keys.id, details.feedpurpose),
+      feedformat: BB1,
+    }
+    const shardVisit = detailsToVisit(shardDetails)
+    find(v1Feed, shardVisit, cb)
+  }
+
+  function findOrCreateShard(v1Feed, root, details, cb) {
```

Comment:
`<style>`

I would recommend shuffling the args to match the path _vibe_ of `root/v1/:shard/subfeedDetails`, i.e.

```js
function findOrCreateShard(root, v1Feed ,details, cb) {
```

Reply:
Yeah that's a nice one.